### ISSUE TITLE
flutter pub publish is very slow / failing

### ISSUE THREAD

**Issue by lukepighetti** [flutter/flutter#32307]
I am trying to publish a previously published package (`hydrated`) to pub using `flutter pub publish` on my iMac and it's taking forever. Took 1 second to show the tree, 30 seconds to show a warning (rename "docs" to "doc", no thanks), and it has been four minutes and won't proceed. Tried this twice.

<img width="606" alt="Screen Shot 2019-05-08 at 7 15 32 AM" src="https://user-images.githubusercontent.com/666539/57371583-133c5c00-7161-11e9-8165-2b60466e64ee.png">

**Comment by lukepighetti** [comment]
Interestingly enough, `pub publish` works perfectly.

**Comment by shunkakinoki** [comment]
Having the same problem here too.

**Comment by d-wolf** [comment]
Same here. I have tried it several times and even after half an hour nothing happened.

````
ESYS-iMac:esys-flutter-share danielwolf$ flutter doctor -v
[✓] Flutter (Channel stable, v1.5.4-hotfix.2, on Mac OS X 10.14.5 18F132, locale en-DE)
    • Flutter version 1.5.4-hotfix.2 at /Users/danielwolf/Documents/GitHub/flutter
    • Framework revision 7a4c33425d (5 weeks ago), 2019-04-29 11:05:24 -0700
    • Engine revision 52c7a1e849
    • Dart version 2.3.0 (build 2.3.0-dev.0.5 a1668566e5)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/danielwolf/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2, Build version 10E125
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 36.0.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] VS Code (version 1.34.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.1.0

[✓] Connected device (1 available)
    • Iphone6 (esys gmbh) • 642e346635eafe54baf1a8f531fee5e797a93c1a • ios • iOS 12.3.1

• No issues found!
```

**Comment by d-wolf** [comment]
Try ```flutter pub pub publish``` instead of ```flutter pub publish```.
Check out [this comment](https://github.com/flutter/flutter/issues/33302#issuecomment-497180224).

**Comment by shunkakinoki** [comment]
`pub publish` works, as @lukepighetti has suggested
`flutter pub pub publish` also did not work for me

**Comment by michael-cheung-thebus** [comment]
`flutter pub pub publish` works for me

didn't have the patience to see what would happen with `flutter pub publish`, but I did wait significantly longer than a few minutes before giving up.

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, v1.5.4-hotfix.2, on Microsoft Windows [Version 10.0.18362.145], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[√] Android Studio (version 3.3)
[√] Connected device (1 available)

• No issues found!
```

**Comment by JoshLorDeveloper** [comment]
I gave up after 4 hours and it was still stuck after giving suggestions.

**Comment by albo1337** [comment]
> 
> 
> Try `flutter pub pub publish` instead of `flutter pub publish`.
> Check out [this comment](https://github.com/flutter/flutter/issues/33302#issuecomment-497180224).

you deserve the crown sir

**Comment by amreniouinnovent** [comment]
Yes flutter pub pub publish works but why pub pub!!!!

**Comment by Bowman74** [comment]
I encountered the same issue. Using  flutter pub pub publish worked for me as well.

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/38490 was merged in 1.9.2.  Please try on a newer version of Flutter.
